### PR TITLE
MessagePolicy in Conversations

### PR DESCRIPTION
**Pull request by javipacheco**
This PR adds new fields in the `MessagePolicy` for managing the way to add messages to the memory and get messages from the memory when we are creating new prompts

### CODE REVIEW

File: core/src/commonMain/kotlin/com/xebia/functional/xef/metrics/LogsMetric.kt [hunk 1]
```diff
@@ -16,13 +16,14 @@ class LogsMetric : Metric {
     val milis = getTimeMillis()
     val name = prompt.messages.lastOrNull()?.content ?: "empty"
     println("Prompt-Span: $name")
+    println("${writeIdent()}|-- Conversation Id: ${conversation.conversationId?.value ?: "empty"}")
```
Comment (Montagon) [inline core/src/commonMain/kotlin/com/xebia/functional/xef/metrics/LogsMetric.kt:19]: I think we should use some kind of logger for this.

File: core/src/commonMain/kotlin/com/xebia/functional/xef/metrics/LogsMetric.kt [hunk 2]
```diff
@@ -16,13 +16,14 @@ class LogsMetric : Metric {
...
     val output = block()
     println("${writeIdent()}|-- Finished in ${getTimeMillis()-milis} ms")
     return output
   }
 
   override fun log(conversation: Conversation, message: String) {
-    println("${writeIdent()}|-- $message".padStart(identSize, ' '))
+    println("${writeIdent()}|-- $message")
```
Comment (Montagon) [inline core/src/commonMain/kotlin/com/xebia/functional/xef/metrics/LogsMetric.kt:26]: Same here

File: examples/kotlin/src/main/kotlin/com/xebia/functional/xef/conversation/conversations/Animal.kt [hunk 3]
```diff
@@ -20,20 +23,58 @@ suspend fun main() {
...
+    val invention: Invention =
+      prompt(
+        Prompt { +user("A groundbreaking invention from the 20th century.") }
+          .copy(configuration = configNoneFromConversation)
+      )
+
+    println()
     println("Animal: $animal")
     println("Invention: $invention")
+    println()
```
Comment (Montagon) [inline examples/kotlin/src/main/kotlin/com/xebia/functional/xef/conversation/conversations/Animal.kt:46]:
```suggestion
    println("\nAnimal: $animal\n")
    println("\nInvention: $invention\n")
```

File: examples/kotlin/src/main/kotlin/com/xebia/functional/xef/conversation/conversations/Animal.kt [hunk 4]
```diff
@@ -20,20 +23,58 @@ suspend fun main() {
...
+        )
 
     val story: String = promptMessage(storyPrompt)
 
+    println()
+    println("Story 1:")
+    println()
     println(story)
+    println()
+    println()
```
Comment (Montagon) [inline examples/kotlin/src/main/kotlin/com/xebia/functional/xef/conversation/conversations/Animal.kt:67]:
```suggestion
    println("\nStory 1:\n $story\n\n")
```

**Review by Montagon — APPROVED** [review]
Just a doubt and minor minor comments. Great job @javipacheco, great approach!

**Review by raulraja — APPROVED** [review]
Looks good @javipacheco , thanks!

**Comment by javipacheco** [comment]
@Montagon We are going to address the logging comments in another PR. Thanks!

**Pull request merged**